fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerCreateCoachNote.js

```diff
@@ -0,0 +1,120 @@
+
+    useEffect(() => {
+        const fetchToken = async () => {
+            const token = await getAccessToken();
+            if (token) setAccessToken(token);
+        };
+
+        fetchToken();
+    }, [getAccessToken]);
+
+    useEffect(() => {
+        if (accessToken) {
+            fetchAllAccounts();
+        }
+    }, [accessToken]);
+
+    const fetchAllAccounts = async () => {
+        try {
+            const response = await fetch(`${process.env.REACT_APP_BASE_URL}/api/v1/accounts`, {
+                method: "GET",
+                headers: {
+                    Authorization: `Bearer ${accessToken}`,
+                    'Content-Type': 'application/json'
+                }
+            });
+            const data = await response.json();
+            if (!response.ok) throw new Error('Network response was not ok');
+            setAccounts(data);
+        } catch (error) {
+            console.error("Error fetching accounts:", error);
+        }
+    };
+
+    const handleInputChange = (e) => {
+        setNewCoachNote({ ...newCoachNote, [e.target.name]: e.target.value });
+    };
+
+    const handleUsernameChange = (e) => {
+        const selectedUsername = e.target.value;
+        const selectedAccount = accounts.find(account => account.username === selectedUsername);
+        setNewCoachNote({
+            ...newCoachNote,
+            username: selectedUsername,
+            accountId: selectedAccount ? selectedAccount.accountId : '',
+            userId: selectedAccount ? selectedAccount.userId : ''
+        });
+    };
+
+    const createCoachNote = async (event) => {
+        event.preventDefault();
+        const coachNoteData = {
+            accountId: newCoachNote.accountId,
+            userId: newCoachNote.userId,
+            username: newCoachNote.username,
+            content_EN: newCoachNote.content_EN,
+            content_FR: newCoachNote.content_FR,
+        };
+        try {
+            const response = await fetch(`${process.env.REACT_APP_BASE_URL}/api/v1/coachnotes`, {
+                method: "POST",
+                headers: {
+                    Authorization: `Bearer ${accessToken}`,
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(coachNoteData)
+            });
+            if (!response.ok) throw new Error('Network response was not ok');
+            navigate("/trainerCoachNotes");
+        } catch (error) {
+            console.error('Error creating coach note:', error);
+        }
+    };
+
+    return (
+        <div className="admin-invoices">
+            <form onSubmit={createCoachNote}>
+                {/* Username Dropdown */}
+                <label htmlFor="username">{t('Username:')}</label>
+                <select id="username" name="username" value={newCoachNote.username} onChange={handleUsernameChange}>
+                    <option value="">Select a username</option>
```

Comment:
Perhaps we need localization here?

Reply:
Yes we would need it will add it on my next pr for the Update Coach Notes.